src/EditorFeatures/CSharpTest/MakeLocalFunctionStatic/MakeLocalFunctionStaticTests.cs

```diff
@@ -191,26 +207,156 @@ void M()
+        {
+            await TestInRegularAndScriptAsync(
+$@"using System;
+
+class C
+{{
+    void M()
+    {{{leadingTrivia}
         int [||]fibonacci(int n)
+        {{
+            return n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
+        }}
+    }}
+}}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        static int fibonacci(int n)
         {
             return n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
         }
     }
 }",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        [InlineData("")]
+        [InlineData("\r\n")]
+        [InlineData("\r\n\r\n")]
+        public async Task TestLeadingTriviaAfterLocalFunction(string leadingTrivia)
+        {
+            await TestInRegularAndScriptAsync(
+$@"using System;
+
+class C
+{{
+    void M()
+    {{
+        bool otherFunction()
+        {{
+            return true;
+        }}{leadingTrivia}
+        int [||]fibonacci(int n)
+        {{
+            return n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
+        }}
+    }}
+}}",
 @"using System;
 
 class C
 {
     void M()
     {
-        int x;
+        bool otherFunction()
+        {
+            return true;
+        }
 
         static int fibonacci(int n)
         {
             return n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
         }
     }
 }",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        [InlineData("")]
+        [InlineData("\r\n")]
+        [InlineData("\r\n\r\n")]
+        public async Task TestLeadingTriviaAfterExpressionBodyLocalFunction(string leadingTrivia)
+        {
+            await TestInRegularAndScriptAsync(
+$@"using System;
+
+class C
+{{
+    void M()
+    {{
+        bool otherFunction() => true;{leadingTrivia}
+        int [||]fibonacci(int n) => n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
+    }}
+}}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        bool otherFunction() => true;
+
+        static int fibonacci(int n) => n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
+    }
+}",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        [InlineData("")]
+        [InlineData("\r\n")]
+        [InlineData("\r\n\r\n")]
+        public async Task TestLeadingTriviaAfterComment(string leadingTrivia)
+        {
+            await TestInRegularAndScriptAsync(
+$@"using System;
+
+class C
+{{
+    void M()
+    {{
+        //Local function comment{leadingTrivia}
+        int [||]fibonacci(int n)
+        {{
+            return n <= 1 ? n : fibonacci(n - 1) + fibonacci(n - 2);
+        }}
+    }}
+}}",
+$@"using System;
+
+class C
+{{
+    void M()
+    {{
+        //Local function comment{leadingTrivia}
```

Comment:
Probably worth noting that no changes are made to blank lines when the local function is preceded by a comment, meaning that all blank lines will be preserved, and no blank lines will be inserted. I confirmed that normal method declarations have the same behavior.